master/internal/db/postgres_tasks_intg_test.go

```diff
@@ -372,3 +386,312 @@ func TestExhaustiveEnums(t *testing.T) {
+
+	err = db.UpdateAllocationStartTime(*aIn)
+	require.NoError(t, err)
+
+	a, err = db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn.StartTime, a.StartTime)
+
+	// Testing UpdateAllocationProxyAddress
+	proxyAddr := "here"
+	aIn.ProxyAddress = &proxyAddr
+
+	err = db.UpdateAllocationProxyAddress(*aIn)
+	require.NoError(t, err)
+
+	a, err = db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err)
+	require.Equal(t, aIn.ProxyAddress, a.ProxyAddress)
+}
+
+func TestCloseOpenAllocations(t *testing.T) {
+	// Create test allocations, with a NULL end time.
+	t1In := RequireMockTask(t, db, nil)
+	a1In := RequireMockAllocation(t, db, t1In.TaskID)
+
+	t2In := RequireMockTask(t, db, nil)
+	a2In := RequireMockAllocation(t, db, t2In.TaskID)
+
+	// Set status for both open allocation as 'terminated'.
+	terminated := model.AllocationStateTerminated
+	a1In.State = &terminated
+	a2In.State = &terminated
+
+	// Close only a2In open allocations (filter out the rest).
+	err := db.CloseOpenAllocations([]model.AllocationID{a1In.AllocationID})
+	require.NoError(t, err)
+
+	a1, err := db.AllocationByID(a1In.AllocationID)
+	require.NoError(t, err)
+	require.Nil(t, a1.EndTime)
+
+	a2, err := db.AllocationByID(a2In.AllocationID)
+	require.NoError(t, err)
+	require.NotNil(t, a2.EndTime)
+
+	// Close the rest of the open allocations.
+	err = db.CloseOpenAllocations([]model.AllocationID{})
+	require.NoError(t, err)
+
+	a1, err = db.AllocationByID(a1In.AllocationID)
+	require.NoError(t, err)
+	require.NotNil(t, a1.EndTime)
+}
+
+func TestTaskLogsFlow(t *testing.T) {
+	t1In := RequireMockTask(t, db, nil)
+	t2In := RequireMockTask(t, db, nil)
+
+	// Test AddTaskLogs & TaskLogCounts
+	taskLog1 := RequireMockTaskLog(t, db, t1In.TaskID, "1")
+	taskLog2 := RequireMockTaskLog(t, db, t1In.TaskID, "2")
+	taskLog3 := RequireMockTaskLog(t, db, t2In.TaskID, "3")
+
+	// Try adding only taskLog1, and count only 1 log.
+	err := db.AddTaskLogs([]*model.TaskLog{taskLog1})
+	require.NoError(t, err)
+
+	count, err := db.TaskLogsCount(t1In.TaskID, []api.Filter{})
+	require.NoError(t, err)
+	require.Equal(t, count, 1)
+
+	// Try adding the rest of the Task logs, and count 2 for t1In.TaskID, and 1 for t2In.TaskID
+	err = db.AddTaskLogs([]*model.TaskLog{taskLog2, taskLog3})
+	require.NoError(t, err)
+
+	count, err = db.TaskLogsCount(t1In.TaskID, []api.Filter{})
+	require.NoError(t, err)
+	require.Equal(t, count, 2)
+
+	count, err = db.TaskLogsCount(t2In.TaskID, []api.Filter{})
+	require.NoError(t, err)
+	require.Equal(t, count, 1)
+
+	// Test TaskLogsFields.
+	resp, err := db.TaskLogsFields(t1In.TaskID)
+	require.NoError(t, err)
+	require.ElementsMatch(t, resp.AgentIds, []string{"testing-agent-1", "testing-agent-2"})
+	require.ElementsMatch(t, resp.ContainerIds, []string{"1", "2"})
+
+	// Test TaskLogs.
+	// Get 1 task log matching t1In task ID.
+	logs, _, err := db.TaskLogs(t1In.TaskID, 1, []api.Filter{}, apiv1.OrderBy_ORDER_BY_UNSPECIFIED, nil)
```

Comment:
Can we test Tasklogs with a few api.Filters to see if we are getting the expected results?

Reply:
added a test block